Prox/Prox/Utilities/OpenInHelper.swift

```diff
@@ -0,0 +1,168 @@
+        }
+    }
+
+    //MARK: Open URL in Browser
+    fileprivate static func openURLInBrowser(url: URL) -> Bool {
+        // check to see if Firefox is available
+        // Open in Firefox or Safari
+        let controller = OpenInFirefoxControllerSwift()
+        if !(controller.isFirefoxInstalled() && controller.openInFirefox(url)) {
+            return UIApplication.shared.openURL(url)
+        }
+
+        return true
+    }
+
+    //MARK: Open URL general
+    fileprivate static func openURL(url: URL) -> Bool {
+        // check to see if we can open in Google Maps
+        if UIApplication.shared.canOpenURL(url) {
+            return UIApplication.shared.openURL(url)
+        }
+
+        return false
+    }
+
+    fileprivate static func open(url: URL, withAppScheme appScheme: String) -> Bool {
+        guard let appSchemeURL = URL(string: appScheme),
+            UIApplication.shared.canOpenURL(appSchemeURL),
+            openURL(url: url) else {
+                return openURLInBrowser(url: url)
+        }
+
+        return true
+    }
+
+//    //MARK: Open Reviews
+//    static func openInYelp(url: URL) -> Bool {
+//        return open(url: url, withAppScheme: yelpAppURLSchemeString)
+//    }
+//
+//    static func openInTripAdvisor(url: URL) -> Bool {
+//        return open(url: url, withAppScheme: tripAdvisorAppURLSchemeString)
+//    }
+//
+//    //MARK: Open Wikipedia
+//    static func openInWikipedia(url: URL) -> Bool {
+//        return open(url: url, withAppScheme: wikipediaAppURLSchemeString)
+//    }
+
+
+    //MARK: Open route in maps
+
+    static func openRoute(fromLocation: CLLocationCoordinate2D, toPlace place: Place, by transportType: MKDirectionsTransportType) -> Bool {
+        // try and open in Google Maps app
+        if let gmapsRoutingRequestURL = gmapsAppURLForRoute(fromLocation: fromLocation, toLocation: place.latLong, by: transportType),
+            openURL(url: gmapsRoutingRequestURL) {
+            return true
+        // open in Apple maps app
+        } else if let address = place.address,
+            let appleMapsRoutingRequest = appleMapsURLForRoute(fromLocation: fromLocation, toAddress: address, by: transportType),
+            openURL(url: appleMapsRoutingRequest)
+        {
+            return true
+        // open google maps in a browser
+        } else if let gmapsRoutingRequestURL = gmapsWebURLForRoute(fromLocation: fromLocation, toLocation: place.latLong, by: transportType),
+            openURLInBrowser(url: gmapsRoutingRequestURL) {
+            return true
+        }
+        print("Unable to open directions")
+        return false
+    }
+
+    fileprivate static func gmapsAppURLForRoute(fromLocation: CLLocationCoordinate2D, toLocation: CLLocationCoordinate2D, by transportType: MKDirectionsTransportType) -> URL? {
+        let directionsMode: String
+        switch transportType {
+        case MKDirectionsTransportType.automobile:
```

Comment:
nit: I wonder if we can better encapsulate all these verbose switches more cleanly. Perhaps we can make an extension function on `MKDirectionsTransportType` like `getGmapsAppDirectionsMode` (and similar for the other functions)